### DISCUSSION TITLE
Alarms don't set off in most of the apps in devices like Vivo, OPPO etc because of doze mode

### DISCUSSION THREAD

**Discussion by CaptainColdAJ** [WrichikBasu/ShakeAlarmClock#61] [General]
Alarms apps when installed in devices like OPPO, VIVO etc sometimes doesn't set off alarms due to their battery optimisations and doze mode. Even if the app is locked in recents, disabled battery optimisations (both system and UI's own) and with AutoStart enabled. Have this app covered this particular area to avoid these kind of problems. If so, what has to be done in the code to make it work, even when the device is in deep sleep mode(doze mode)

**Comment by WrichikBasu** [comment]
Thanks for your interest in the app.

Deep sleep is a problem, I agree. And unfortunately, not much documentation is available on this topic. I have implemented an option to request users to disable battery optimizations. In addition, this app has all the required permissions declared to make it stand out as an alarm app, which will tell the system to get out of doze mode when an alarm is about to ring. Other than these, I really can't do much unless I find documentation.

I don't own Oppo or Vivo devices, so cannot confirm what their OEM is doing. But it is very well possible that they have implemented a more severe battery optimization policy. And the vendors can make their own apps immune from these optimizations.

If you find something specific to these OEMs regarding their battery optimization strategies and how to save an app in such a situation, do let me know, and I will try to implement it.

Wrichik.
